meta-mbl-distro/recipes-core/base-files/base-files_%.bbappend

```diff
@@ -42,6 +42,27 @@ python __anonymous() {
         d.appendVar("FSTAB_LINES", fstab_line)
 }
 
+# Write the partition sizes and offsets to files which will be placed in /config/factory
+python __anonymous() {
+    import pathlib
+
+    path_prefix = d.getVar("WORKDIR")
+    # the WORKDIR doesn't exist at recipe parse-time, so create it.
+    pathlib.Path(path_prefix).mkdir(parents=True, exist_ok=True)
+    mbl_partition_data_files = "MBL_PARTITION_DATA_FILES"
+    data_required = ["align_KiB", "offsets_KiB", "size_KiB"]
+
+    for part_info in d.getVar("MBL_PARTITION_INFOS"):
+        if part_info["skip"]:
+            continue
+
+        for data_item in data_required:
+            fname = "{}_{}".format(part_info["name"], data_item)
+            fpath = pathlib.Path(path_prefix, fname)
+            fpath.write_text(str(part_info[data_item]))
```

Comment:
What happens here when `data_item` is  `"offsets_KiB"`? `part_info["offsets_KiB"]` is a `list`...

Reply:
It saves a string representation of the list, which ends up being "[item, item2]" etc. I guess I could save each element on it's own line.

Reply:
Yeah, I think being able to parse the data very easily should be a high priority here.